apps/designer/app/designer/shared/tree/index.tsx

```diff
@@ -15,20 +15,24 @@ const instanceRelatedProps = {
   getItemPath: utils.tree.getInstancePath,
   getItemPathWithPositions: utils.tree.getInstancePathWithPositions,
   canLeaveParent(item: Instance) {
-    return getComponentMeta(item.component).isInlineOnly !== true;
+    const { type } = getComponentMeta(item.component);
+    return (
+      type === "container" ||
+      type === "control" ||
+      type === "embed" ||
+      type === "rich-text"
```

Comment:
isn't this basically type !== 'rich-text-child'?

Reply:
Yes, but when/if new type is added we will need to catch all such cases. It's also easier to understand when behaviour is applied rather than when it's not.

Reply:
true, though I still wonder if it would be better to simplify this, maybe by adding a utility function? just wondering if its gonna be easy to overlook one usage, especially when there are multiple of them, once we need to add another case

Reply:
So, when a new type is added, unless this code is updated it won't be able to live parent by default. Not sure, but I think default should be opposite.

Also, whatever we decide to keep here, it should match what we do here: https://github.com/webstudio-is/webstudio-designer/blob/9ed4d4a18bad75c92d71315c5106b13127eefc96/apps/designer/app/canvas/shared/use-drag-drop.ts#L181-L182